src/ptc/util/jms.clj

```diff
@@ -75,6 +75,30 @@
          (map key->key)
          (into {}))))
 
+(defn update-cloud-path-keys
+  "If a file exists at the JMS cloud path, add the key to 'copy' otherwise
+  add the JMS key for the on-prem path to 'push'"
+  [wfl-key key-map workflow]
+  (let [[jms-cloud-key jms-on-prem-key] (get-in key-map [::push wfl-key])
+        on-prem-path (get workflow jms-on-prem-key)
+        cloud-path (get workflow jms-cloud-key)
+        uploaded? (try
+                 (misc/shell! "gsutil" "stat" cloud-path)
```

Comment:
I think this duplicates code in `wait-for-files-in-bucket`; it would be nice to avoid the code duplication if possible.

Reply:
I think wait for files in bucket assumes that the files should be there and it will actually block forever until that happens. It doesn't do any timing out itself so we could just give it a very short timeout here to reuse the code if we want.

Reply:
These lines of code
```
(try
                       (misc/shell! "gsutil" "stat" file)
                       (catch Exception _
                         nil))
```
are exactly the same, as far as I can tell. Putting it into a method like `gcs-exists` would avoid the duplication.